src/battle_script_commands.c

```diff
@@ -16212,10 +16212,14 @@ void BS_SetRemoveTerrain(void)
     else
     {
         u16 atkHoldEffect = GetBattlerHoldEffect(gBattlerAttacker, TRUE);
+        u8 varTerrainTimer = VarGet(B_VAR_TERRAIN_TIMER);
```

Comment:
Did you add the var check in `BS_SetRemoveTerrain` to allow for custom timers on terrain abilities / moves?

Reply:
Yes, it's just a few lines below. This review comment you made *is* for a line from `BS_SetRemoveTerrain` 😂

Reply:
What happens if say `B_VAR_TERRAIN_TIMER` was set to 10 to work for `B_VAR_TERRAIN` but you would want to preserve the old terrain abiliit / move timer?

Reply:
`B_VAR_SET_TERRAIN_TIMER` would take preference if it has a value of 1 or higher.

I'll add a `&& VarGet(B_VAR_TERRAIN)` real quickly and test to make sure nothing breaks.

Reply:
If you add `&& VarGet(B_VAR_TERRAIN)` does it make sense to keep it at all?

Reply:
I'm not following 😅 

You're telling me to perform this change?

```diff
-       if (varTerrainTimer)
+       if (varTerrainTimer && !VarGet(B_VAR_TERRAIN))
```

Because it doesn't make sense to me.

I could do

```diff
-       if (varTerrainTimer)
+       if (!varTerrainTimer && !VarGet(B_VAR_TERRAIN))
```

though. That makes sense and it looks nice and clean as well :eyes:

Reply:
To elaborate, that would basically be:

```diff
-       if (varTerrainTimer)
-           gFieldTimers.terrainTimer = varTerrainTimer;
-       else
-           gFieldTimers.terrainTimer = (atkHoldEffect == HOLD_EFFECT_TERRAIN_EXTENDER) ? 8 : 5;
+       if (!varTerrainTimer && !VarGet(B_VAR_TERRAIN))
+           gFieldTimers.terrainTimer = (atkHoldEffect == HOLD_EFFECT_TERRAIN_EXTENDER) ? 8 : 5;
+       else
+           gFieldTimers.terrainTimer = varTerrainTimer;
```

Reply:
> I'm not following 😅
> 
> You're telling me to perform this change?
> 
> ```diff
> -       if (varTerrainTimer)
> +       if (varTerrainTimer && !VarGet(B_VAR_TERRAIN))
> ```
> 
> Because it doesn't make sense to me.

You are right. This doesn't make sense lol

> I could do
> 
> ```diff
> -       if (varTerrainTimer)
> +       if (!varTerrainTimer && !VarGet(B_VAR_TERRAIN))
> ```
> 
> though. That makes sense and it looks nice and clean as well 👀
I don't think this solves the problem. Afaik `B_VAR_TERRAIN` isn't reset mid-turn.

Reply:
>  I don't think this solves the problem. Afaik B_VAR_TERRAIN isn't reset mid-turn.

This seems to work.

```diff
diff --git a/src/battle_script_commands.c b/src/battle_script_commands.c
index 37397c33d..04e7720c7 100644
--- a/src/battle_script_commands.c
+++ b/src/battle_script_commands.c
@@ -16212,11 +16212,12 @@ void BS_SetRemoveTerrain(void)
     else
     {
         u16 atkHoldEffect = GetBattlerHoldEffect(gBattlerAttacker, TRUE);
+        u16 varTerrain = VarGet(B_VAR_TERRAIN);
         u8 varTerrainTimer = VarGet(B_VAR_TERRAIN_TIMER);

         gFieldStatuses &= ~STATUS_FIELD_TERRAIN_ANY;
         gFieldStatuses |= statusFlag;
-        if (varTerrainTimer)
+        if (varTerrainTimer && varTerrain && gFieldStatuses & varTerrain)
             gFieldTimers.terrainTimer = varTerrainTimer;
         else
             gFieldTimers.terrainTimer = (atkHoldEffect == HOLD_EFFECT_TERRAIN_EXTENDER) ? 8 : 5;
lunos:/mnt/c/Users/Lunos/Home/rhh$
```

Reply:
Resolved in Discord